server/api/user_handler.py

```diff
@@ -102,10 +99,8 @@ def edit_profile():
 
 @user_handler.route('/api/get_user', methods=['GET'])
 @require_auth
-def get_user_profile():
-    token = request.cookies.get("auth_token")
-    data = jwt.decode(token, app.app.config['JWT_SECRET'], algorithms=['HS256'])
-    current_user = User.query.filter_by(username=data['user']).first()
+@get_current_user
```

Comment:
looks like this branch is a little out dated bcause it includes all the middleware things. be sure to update your branch before you merge this in :)

Reply:
It included updated middleware changes, I think It can be merged into dev